### PR TITLE
4081 - Fix missing data for excel export with datagrid

### LINKED ISSUE #4081: Excel export: The excel exported from DataGrid with group have no data.
**Describe the bug**
We have upgraded ids-enterprise to 4.28.0 and tried to export excel from datagrid with group, but the generated excel had no data, just a header. But if no group, the excel will have data.  Both are normal with data when using infor/sohoxi 4.5.0

**To Reproduce**
1. Access the example: [https://usspvwhvgmdemo:20493/ids-enterprise-ng-demo/](url)
2. Click on "D→DataGrid - Groups" on the left menu, and the page shows a table with group and data.
3. Click on the link "Export to Excel".
4. Open the exported excel, you can see it only includes group info and other data is missing. Please refer to the screenshots.

**Expected behavior**
The exported excel should include all data in the DataGrid.

**Version**

- ids-enterprise: [v4.28.0 or v4.29.0]

**Screenshots**
The DataGrid page:
image

The exported excel:
image

**Platform**
All

**Additional context**
DataGrid-groupable demo code:
[datagrid-groupable.demo.zip](https://github.com/infor-design/enterprise/files/4824610/datagrid-groupable.demo.zip)

### PR DESCRIPTION
**Pull request by deep7102**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**
Fixed data was not exporting to excel when use settings groupable with Datagrid.

**Related github/jira issue (required)**:
Closes #4081

**Steps necessary to review your pull request (required)**:
- Pull this branch and build/run the demo app
- Navigate to: http://localhost:4000/components/datagrid/test-grouping-export.html
- Click button `Export To Excel`
- It should download to your local machine excel file `myExport.xls`
- Open this file and see the exported data, it will be ungrouped (flat data).

**Included in this Pull Request**:
- [x] A note to the change log.

### CODE REVIEW

**Comment by tmcconechy** [comment]
@deep7102 Did you happen to go back to the past version this worked on? Wondering if it should show the groups or not?

**Comment by deep7102** [comment]
@tmcconechy 
Yes, I tried and looks it was never worked or broken since long time.

https://4130-enterprise.demo.design.infor.com/components/datagrid/example-grouping.html
- Open developer tools and run
- const grid = $('#datagrid').data('datagrid');
- Soho.excel.exportToExcel(false, false, false, grid);
- It will download excel file to local
- Looking in this generated file does not have all data

**Comment by tmcconechy** [comment]
Ok i see. And the old version it did it without the groups but was missing data. I'm think maybe we add an option "includeGroups". In the export settings and they either show it grouped as on page or ungrouped?

**Comment by GauravJags** [comment]
Is the Issue resolved for SohoDataGrid?

**Comment by tmcconechy** [comment]
Almost @GauravJags - this PR is in progress and under review. Actually please answer me one question?
Would you like to see the groups in the export or is it ok if it comes out flat?

Thanks @GauravJags

**Comment by deep7102** [comment]
Hi @tmcconechy updated now, to export as ungrouped (flat data).

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 - `[Application Menu]` Fixed a bug where some buttons did not have labels for the icon buttons in toolbars. Check your application if you use this pattern. ([#4085](https://github.com/infor-design/enterprise/issues/4085))
 - `[Buttons]` Reverted an inner Css rule change that set 'btn' classes to contains vs starts with. ([#4120](https://github.com/infor-design/enterprise/issues/4120))
 - `[Datagrid]` Fixed an issue where the tooltip for tree grid was not working properly. ([#827](https://github.com/infor-design/enterprise-ng/issues/827))
+- `[Datagrid]` Fixed an issue where the data was not exporting to excel when use settings groupable. ([#4081](https://github.com/infor-design/enterprise/issues/4081))
```
Comment (tmcconechy) [inline docs/CHANGELOG.md:16]:
```suggestion
- `[Datagrid]` Fixed an issue where the data was not exporting to excel when using the groupable setting. ([#4081](https://github.com/infor-design/enterprise/issues/4081))
```

**Pull request merged**